actionview/lib/action_view/template_eager_loader.rb

```diff
@@ -0,0 +1,123 @@
+module ActionView
+  # = Action View Template Eager Loader
+  class TemplateEagerLoader
+    # ActionView::TemplateEagerLoader finds and caches all
+    # view templates and then compiles them.
+    #
+    # If this is done on server startup,
+    # the first request will be faster.
+
+    def initialize(resolver)
+      @resolver = resolver
+    end
+
+    # Compiles and caches all templates
+    def eager_load
+      each_template(&:compile!)
+    end
+
+    private
+
+    # Determines all possible template parameters and then
+    # delegates finding and caching to the Resolver.
+    # After finding and caching the templates, it will
+    # yield them to the block.
+    def each_template(&block)
+      views.each do |prefix, name|
+        locales.each do |locale|
+          details, key, action, partial, locals = set_template_parameters(locale, name)
+          locals.each do |locals_array|
+            @resolver.find_all(action, prefix, partial, details, key, locals_array).each(&block)
+          end
+        end
+      end
+    end
+
+    def set_template_parameters(locale, name)
+      name_parts, partial = name_parts_and_partial(name)
+      details, key = details_and_key(locale, name, name_parts)
+      action = partial ? name_parts.first[1..-1] : name_parts.first
```

Comment:
`name_parts.first` isn't an array but a string. This part removes an underscore from the beginning of the partial name, e.g. '_partial' would become 'partial'.